Manage the syslog-ng repo

smortex opened the pull request:
Fresh packages are available for Debian and RedHat based systems.  Add
support for these external repositories and make it possible to skip
using them.

smortex committed:
Manage the syslog-ng repo

Fresh packages are available for Debian and RedHat based systems.  Add
support for these external repositories and make it possible to skip
using them.

smortex committed:
Request stdlib >= 4.6.0

assert_private() was introduced in stdlib 4.6.0

faxm0dem commented on manifests/init.pp:
```diff
@@ -11,6 +11,7 @@
   $package_ensure,
   $manage_init_defaults = false,
   $manage_package       = true,
+  $manage_repo          = false,
```
I think setting this will render the data/ defaults useless.
So maybe remove the false here and put it into data/default.yaml

smortex replied:
It seems to be working in production, but it hides the fact that some systems will default to `true`… 

I will work on this today.

FYI, there is discussion related to this ATM at voxpupuli (PR43 of voxpupuli/puppet-gitlab_ci_runner… not putting an URL to avoid this issue being linked there: it would not bring new information in the context of this distant PR)

smortex replied:
I have amended the commit to have the default value in Hiera.

faxm0dem commented on .fixtures.yml:
```diff
@@ -1,8 +1,11 @@
+    apt: 'https://github.com/puppetlabs/puppetlabs-apt.git'
+    concat: 'https://github.com/puppetlabs/puppetlabs-concat.git'
+    stdlib: 'https://github.com/puppetlabs/puppetlabs-stdlib.git'
+    translate: 'https://github.com/puppetlabs/puppetlabs-translate.git'
```
why is translate needed?

smortex replied:
It's a [dependency needed by apt](https://github.com/puppetlabs/puppetlabs-apt/blob/master/metadata.json#L15-L18).  I did not added apt / yumrepo to the `metadata.json` file (only to the fixture files) in order to not pull-in new dependencies.

Let me know if you prefer the module to depend hardly on this "optional" dependencies.

faxm0dem replied:
Understood, it's fine then

faxm0dem reviewed (commented):
thanks this is a good idea!

I think what's missing is documentation, and also a comment on the fact that those are unofficial repos, even though czanik and lazlo work at balabit^woneidentity

faxm0dem commented:
If you have the time please add a word or two to the documentation about the repo.

smortex committed:
Document $syslog_ng::manage_repo

Document the default behaviour and the list of things that the module
manages.

smortex commented:
I updated the README to document the new behavior as requested.  Sorry, I missed your comment (twice) :confused:

faxm0dem commented:
one last thing: we have two choices here concerning semantic versioning:

1. we keep repo management to false and update minor version
2. we set it to true, but update major version (3.x)

My opinion would be to do 1., as there aren't so many changes since 2.x that justify a major version change

smortex commented:
Yep, I was thinking about this when documenting the new behaviour…

I rebased my changes on top of master, amended the commits to default to false, added some tests to ensure that when setting `manage_repo` to `true` it works as expected, and updated the documentation accordingly.

smortex committed:
Run "apt-get update" before installing packages

When adding a repository on a Debian-based system, it's important to
have 'apt-get update' run _before_ installing packages.  Adding the
source notifies 'apt-get update', so we just have to ensure that all
packages managed by the module are managed after 'apt-get update' has
run.

faxm0dem commented:
Thanks for your work on this!
